src/Shared/Extensions/PackageUrlExtension.cs

```diff
@@ -40,6 +41,21 @@ public static bool HasNamespace(this PackageURL packageUrl)
         return packageUrl.Namespace.IsNotBlank();
     }
     
+    /// <summary>
+    /// We want npm package's namespace to be prefixed with "%40" the percent code of "@".
+    /// </summary>
+    /// <param name="packageUrl">The <see cref="PackageURL"/> to get a formatted namespace for.</param>
+    /// <returns>The formatted namespace of <paramref name="packageUrl"/>.</returns>
+    public static string GetNamespaceFormatted(this PackageURL packageUrl)
+    {
+        if (packageUrl.Type != "npm" || packageUrl.Namespace.StartsWith("%40") || packageUrl.Namespace.StartsWith("@"))
+        {
+            return packageUrl.Namespace;
+        }
```

Comment:
I see that this accepts a namespace string starting with `@` - would you mind elaborating on that? Should this method only return a namespace value in one format (encoded or not encoded)?

Reply:
I believe this has been resolved. What the method does now is if the namespace isn't from an NPM package, or if it has `%40` at the beginning of the string, then it just returns the namespace.
If it starts with `@` it removes the `@` at the beginning of the namespace and replaces it with `%40`.
Otherwise, if it is a scoped NPM package and the namespace doesn't start with `@` or `%40` it prepends the namespace with `%40`.